### PR TITLE
10 postCommentsById completed; more error handling added in the app.js

### PR DESCRIPTION
**Pull request by BINQIU1220**

### CODE REVIEW

File: __tests__/comments.test.js [hunk 1]
```diff
@@ -58,3 +58,85 @@ it("status: 400, responds with invalid request message when passed in invalid id
...
+    };
+    return request(app)
+      .post("/api/reviews/1/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
+});
```
Comment (elijaharley) [inline __tests__/comments.test.js:142]: Really excellent testing here Bin, very comprehensive - I'd consider adding a test for if there are unnecessary properties on the body - it should still be successful, but ignore the extra key value pairs.

File: controllers/comments-controller.js [hunk 2]
```diff
@@ -21,3 +24,22 @@ exports.getCommentsById = (req, res, next) => {
...
+    insertCommentsById(id, newComment),
+  ];
+  return Promise.all(promises)
+    .then((data) => {
+      res.status(201).send({ comment: data[2] });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
```
Comment (elijaharley) [inline controllers/comments-controller.js:45]: Do we need to be checking if the property exists before we try and post here or is trying to post to a non-existant endpoint something that we can simply handle as a PSQL error?

**Review by elijaharley — CHANGES_REQUESTED** [review]
Really nice work on this one, Bin - just a couple of things to work on before merging. You can leave this until after the lecture though!

File: __tests__/comments.test.js (hunk 1 above)
Comment (BINQIU1220) [reply __tests__/comments.test.js:164]: Test for ignoring extra keys added and passed as suggested. Thanks Eli.

File: controllers/comments-controller.js [hunk 3]
```diff
@@ -21,3 +24,21 @@ exports.getCommentsById = (req, res, next) => {
...
     });
 };
+
+// TASK 10
+exports.postCommentsById = (req, res, next) => {
+  const id = req.params.review_id;
+  const newComment = req.body;
+
+  const promises = [
+    checkValExists("reviews", "owner", req.body.username),
```
Comment (daviidbartlett) [inline controllers/comments-controller.js:34]: this check is unnecessary, please remove

**Pull request merged**